### PR TITLE
Azure cache

### LINKED ISSUE #2551: Cache dependencies for Azure builds
To speed up build times considerably (as long as cache hits), ref discussion here https://github.com/valhalla/valhalla/pull/2521#issuecomment-678341950

### PR DESCRIPTION
**Pull request by nilsnolde**
fixes #2551 

try out Azure's cache.

## Tasklist

 - [ ] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

### CODE REVIEW

**Comment by nilsnolde** [comment]
Cache works now!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## Release Date: 2020-08-?? Valhalla 3.1.0
 * **Removed**
-   * REMOVED:  Remove Node bindings. [#2502](https://github.com/valhalla/valhalla/pull/2502)
+   * REMOVED:  Remove Node bindings. [#2502](https://github.com/valhalla/valhalla/pull/2502)
```
Comment (kevinkreiser) [inline CHANGELOG.md:3]: typo

File: CHANGELOG.md [hunk 2]
```diff
@@ -85,6 +85,7 @@
    * FIXED: Multicue enter roundabout [#2556](https://github.com/valhalla/valhalla/pull/2556)
 
 * **Enhancement**
+   * ADDED: Dependency cache for Azure [#2567](https://github.com/valhalla/valhalla/pull/2567)
```
Comment (kevinkreiser) [inline CHANGELOG.md:88]: we've been putting these as newest are last. we should probably document that, sorry for the confusion

Comment (nilsnolde) [reply CHANGELOG.md]: Oops, did that for all my PRs.. Should I clean that up?

Comment (kevinkreiser) [reply CHANGELOG.md]: if you want to or if you'd rather wait for the next one no problem. not a huge deal, just letting you know for future reference

Comment (nilsnolde) [reply CHANGELOG.md]: done

**Pull request merged**